File: bindings/pydeck/examples/a5_layer.py
"""
A5Layer
=======

Plot of bike parking density in San Francisco using the A5 geospatial indexing system.

A5 is a pentagonal discrete global grid system that provides equal-area cells worldwide.
Learn more at https://a5geo.org

This example is adapted from the deck.gl documentation.
"""

import pydeck as pdk
import pandas as pd

A5_DATA = "https://raw.githubusercontent.com/visgl/deck.gl-data/master/website/sf.bike.parking.a5.json"

df = pd.read_json(A5_DATA)

# Define a layer to display on a map
layer = pdk.Layer(
    "A5Layer",
    df,
    pickable=True,
    extruded=True,
    get_pentagon="pentagon",
    get_fill_color="[(1 - count / 211) * 235, 255 - 85 * count / 211, 255 - 170 * count / 211]",
    get_elevation="count",
    elevation_scale=10,
)

# Set the viewport location - centered on San Francisco
view_state = pdk.ViewState(
    latitude=37.74,
    longitude=-122.4,
    zoom=11,
    bearing=0,
    pitch=40
)

# Render
r = pdk.Deck(
    layers=[layer],
    initial_view_state=view_state,
    tooltip={"text": "{pentagon} count: {count}"}
)
r.to_html("a5_layer.html")

File: test/render/test-cases/a5-layer.js
// deck.gl
// SPDX-License-Identifier: MIT
// Copyright (c) vis.gl contributors

import {A5Layer} from '@deck.gl/geo-layers';
import {pentagons} from 'deck.gl-test/data';

export default [
  {
    name: 'a5-layer-flat',
    viewState: {
      latitude: 37.75,
      longitude: -122.45,
      zoom: 11,
      pitch: 0,
      bearing: 0
    },
    layers: [
      new A5Layer({
        data: pentagons,
        opacity: 0.8,
        filled: true,
        stroked: false,
        getPentagon: f => f.pentagon,
        getFillColor: f => {
          const value = f.count / 211;
          return [(1 - value) * 235, 255 - 85 * value, 255 - 170 * value];
        },
        pickable: true
      })
    ],
    goldenImage: './test/render/golden-images/a5-layer-flat.png'
  },
  {
    name: 'a5-layer',
    viewState: {
      latitude: 37.78,
      longitude: -122.45,
      zoom: 10.5,
      pitch: 40,
      bearing: 0
    },
    layers: [
      new A5Layer({
        data: pentagons,
        opacity: 0.8,
        filled: true,
        extruded: true,
        getPentagon: f => f.pentagon,
        getFillColor: f => {
          const value = f.count / 211;
          return [(1 - value) * 235, 255 - 85 * value, 255 - 170 * value];
        },
        getElevation: f => f.count,
        elevationScale: 30,
        pickable: true
      })
    ],
    goldenImage: './test/render/golden-images/a5-layer.png'
  }
];
